tests/flows/relayer_modifies_message.go

```diff
@@ -105,7 +107,12 @@ func createAlteredReceiveCrossChainMessageTransaction(
 	numSigners, err := signedMessage.Signature.NumSigners()
 	Expect(err).Should(BeNil())
 
-	gasLimit, err := gasUtils.CalculateReceiveMessageGasLimit(numSigners, requiredGasLimit)
+	gasLimit, err := gasUtils.CalculateReceiveMessageGasLimit(
+		numSigners,
+		requiredGasLimit,
+		len(signedMessage.Bytes()),
```

Comment:
Should this be `len(signedMessage.Payload)`? Otherwise I think we are including the length of the aggregate signature and bit vector, which are not exposed within the EVM.

Reply:
In `subnet-evm` the gas cost per Warp message byte is applied on each predicate byte, which includes the full signed Warp message bytes, so matched that for this estimate. https://github.com/ava-labs/subnet-evm/blob/4a75fa782de9c7b97bbcfb21b0211230234a4481/precompile/contracts/warp/config.go#L144

Reply:
Ah, nice catch. I didn't realize this.